### PR TITLE
fix: Select the right section when categories and tags have the same name in Viz Gallery

### LINKED ISSUE #16012: [gallery] Treemap did not show in #Distribution
Treemap has a Distribution tag but did not show when user select #Distribution or 'Distribution' under category. 

### Expected results

Treemap will show when user select #Distribution or select 'Distribution' under category

### Actual results

Treemap did not show when user select #Distribution or select 'Distribution' under category

#### Screenshots

Screen Shot 2021-08-01 at 9 21 39 PM
Screen Shot 2021-08-01 at 9 32 26 PM

#### How to reproduce the bug

1. Open Gallery, look for Treemap and saw 'Distribution' in the Treemap tag
2. Click #Distribution on the left selection menu and look for Treemap
3. Click 'Distribution' under category on the left selection menu and look for Treemap

### Environment

(please complete the following information):

- superset version: `latest master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR fixes an issue that was causing the Viz Gallery to select the category-related viz types when a tag with the same category name was selected.

Fixes: #16012

### BEFORE

https://user-images.githubusercontent.com/60598000/127860476-b066f95e-f617-48f4-9aba-ef4ae9a6f349.mp4

### AFTER

https://user-images.githubusercontent.com/60598000/127860005-c3aed9a0-6c07-4603-b5fc-671d556dd2f4.mp4

### TESTING INSTRUCTIONS
1. Open the Viz Gallery, look for Treemap and see 'Distribution' in the Treemap tags
2. Click #Distribution on the left selection menu and look for Treemap
3. Click 'Distribution' under category on the left selection menu and look for Treemap

### ADDITIONAL INFORMATION

- [x] Has associated issue: #16012
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
@rosemarie-chiu @jinghua-qa

File: superset-frontend/src/explore/components/controls/VizTypeControl/VizTypeGallery.tsx [hunk 1]
```diff
@@ -52,6 +52,13 @@ type VizEntry = {
   value: ChartMetadata;
 };
 
+enum SECTIONS {
+  ALL_CHARTS = 'ALL_CHARTS',
+  CATEGORY = 'CATEGORY',
+  TAGS = 'TAGS',
+  RECOMMENDED_TAGS = 'RECOMMENDED_TAGS',
+}
```
Comment (villebro) [inline superset-frontend/src/explore/components/controls/VizTypeControl/VizTypeGallery.tsx:60]: nit: I believe the convention is to use Pascal case for enums: https://www.typescriptlang.org/docs/handbook/enums.html

**Review by villebro — APPROVED** [review]
LGTM with minor non-blocking comment

**Review by michael-s-molina — APPROVED** [review]
The code looks good but I would like QA sign-off before merging.

**Comment by junlincc** [comment]
thanks for the fix! when things slow down, I will go through all the Tags, Descriptions and Categories again to make sure they make sense, and are optimized for search. sorry for the trouble 🙏

**Review by jinghua-qa — APPROVED** [review]
LGTM!

https://user-images.githubusercontent.com/81597121/127897628-1e43389e-b582-437b-a2e0-eb1d45a6c4b3.mov

**Review by suddjian — APPROVED** [review]

**Pull request merged**